setup.cfg

```diff
@@ -49,4 +49,63 @@ testpaths = poppy docs
 astropy_header = true
 doctest_plus = enabled
 text_file_format = rst
-addopts = -p no:warnings
\ No newline at end of file
+addopts = -p no:warnings
+
+[coverage:run]
+source = poppy
+omit =
+   poppy/conftest*
+   poppy/cython_version*
+   poppy/setup_package*
+   poppy/*/setup_package*
+   poppy/*/*/setup_package*
+   poppy/tests/*
+   poppy/*/tests/*
+   poppy/*/*/tests/*
+   poppy/version*
+   */poppy/conftest*
+   */poppy/cython_version*
+   */poppy/setup_package*
+   */poppy/*/setup_package*
+   */poppy/*/*/setup_package*
+   */poppy/tests/*
+   */poppy/*/tests/*
+   */poppy/*/*/tests/*
+   */poppy/version*
+
+[coverage:report]
+exclude_lines =
+   # Have to re-enable the standard pragma
+   pragma: no cover
+
+   # Don't complain about packages we have installed
+   except ImportError
+
+   # Don't complain if tests don't hit assertions
+   raise AssertionError
+   raise NotImplementedError
+
+   # Don't complain about script hooks
+   def main\(.*\):
+
+   # Ignore branches that don't pertain to this version of Python
+   pragma: py{ignore_python_version}
+omit =
```

Comment:
It seems annoying & suboptimal that this list of omits has to be repeated twice, but I'll take your word that's how it has to be set up...

Reply:
Yep I agree, but for some reason putting omit in the coverage run isn't enough - it'll still show up in the report.